library/core/src/ptr/non_null.rs

```diff
@@ -279,6 +279,7 @@ impl<T: ?Sized> NonNull<T> {
     #[must_use]
     #[inline]
     #[unstable(feature = "strict_provenance", issue = "95228")]
+    #[ensures(|result| result.get() != 0 && result.get() == self.as_ptr() as *const() as usize)]
```

Comment:
The first conjunct (`result.get() != 0`) should not be needed as it is subsumed by the second conjunct (since `self.as_ptr()` is guaranteed to not be NULL).

Reply:
Also, is the cast to `*const ()` before casting to `usize` necessary?

Reply:
@zhassan-aws Yes otherwise it gives the error `casting `*mut T` as `usize` is invalid`

Reply:
@zhassan-aws Yes otherwise it gives the error `casting `*mut T` as `usize` is invalid`.

Reply:
Can you use the `addr()` method instead?

Reply:
@celinval the function under verification is `addr()`.